app/controllers/visualizations_controller.rb

```diff
@@ -102,15 +102,17 @@ def create
     end
 
     # Try to make a thumbnail
-    mo = MediaObject.new
-    mo.media_type = 'image'
-    mo.name = 'image.png'
-    mo.file = 'image.png'
-    mo.user_id = @cur_user.id
-    mo.check_store!
+    mo = nil
 
     if params[:visualization].try(:[], :svg)
       begin
+        mo = MediaObject.new
+        mo.media_type = 'image'
+        mo.name = 'image.png'
+        mo.file = 'image.png'
+        mo.user_id = @cur_user.id
+        mo.check_store!
+
         image = MiniMagick::Image.read(params[:visualization][:svg], '.svg')
         image.format 'png'
```

Comment:
@asalani93 One thing I noticed about saving a vis that was shown by the imagemagick issue is that if it ever fails here then the vis page will hang trying to save.